src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/TestConnectionManager.kt

```diff
@@ -102,4 +113,739 @@ class TestConnectionManager {
+                        0,
+                        counter.get()
+                    ) { "Unexpected number of messages received. The first message shouldn't be received" }
+                    Thread.sleep(200)
+                    LOGGER.info { "Creating the correct exchange..." }
+                    declareFanoutExchangeWithBinding(it, exchange, queueName)
+                    Thread.sleep(200)
+                    LOGGER.info { "Exchange created!" }
+
+                    Assertions.assertDoesNotThrow {
+                        connectionManager.basicPublish(exchange, "", null, "Hello2".toByteArray(Charsets.UTF_8))
+                    }
+
+                    Thread.sleep(200)
+                    Assertions.assertEquals(
+                        1,
+                        counter.get()
+                    ) { "Unexpected number of messages received. The second message should be received" }
+
+                }
+            }
+    }
+
+    @Test
+    fun `connection manager handles ack timeout`() {
+        val configFilename = "rabbitmq_it.conf"
+        val queueName = "queue4"
+        val prefetchCount = 10
+
+        RabbitMQContainer(DockerImageName.parse(RABBIT_IMAGE_NAME))
+            .withRabbitMQConfig(MountableFile.forClasspathResource(configFilename))
+            .withQueue(queueName)
+            .use {
+                it.start()
+                LOGGER.info { "Started with port ${it.amqpPort}" }
+                val confirmationTimeout = Duration.ofSeconds(1)
+                val counter = AtomicInteger(0)
+                ConnectionManager(
+                    RabbitMQConfiguration(
+                        host = it.host,
+                        vHost = "",
+                        port = it.amqpPort,
+                        username = it.adminUsername,
+                        password = it.adminPassword,
+                    ),
+                    ConnectionManagerConfiguration(
+                        subscriberName = "test",
+                        prefetchCount = prefetchCount,
+                        confirmationTimeout = confirmationTimeout,
+                        minConnectionRecoveryTimeout = 100,
+                        maxConnectionRecoveryTimeout = 200,
+                        maxRecoveryAttempts = 5
+                    ),
+                ).use { connectionManager ->
+                    Thread {
+                        connectionManager.basicConsume(queueName, { _, delivery, ack ->
+                            LOGGER.info { "Received 1 ${delivery.body.toString(Charsets.UTF_8)} from \"${delivery.envelope.routingKey}\"" }
+                            if (counter.get() != 0) {
+                                ack.confirm()
+                                LOGGER.info { "Confirmed!" }
+                            } else {
+                                LOGGER.info { "Left this message unacked" }
+                            }
+                            counter.incrementAndGet()
```

Comment:
We can detect incoming of both messages via CountDownLatch(...). Also we should set 1 for prefetchcount because rabbitMQ doesn't send next messages until confirm for prefetched